docs-linting/src/extension.ts

```diff
@@ -0,0 +1,79 @@
+"use strict";
+
+/* The module 'vscode' contains the VS Code extensibility API
+ The LoadToolbar function will populate items in the toolbar, but only when the extension loads the first time.
+ The common file provides functions that are useful across all commands.
+ Logging, Error Handling, VS Code window updates, etc.
+*/
+
+import { ConfigurationTarget, ExtensionContext, window, workspace } from "vscode";
+import { addFrontMatterTitle } from "./controllers/lint-config-controller";
+import { generateTimestamp } from "./helper/common";
+
+export const output = window.createOutputChannel("docs-linting");
+export let extensionPath: string;
+
+/**
+ * Provides the commands to the extension. This method is called when extension is activated.
+ * Extension is activated the very first time the command is executed.
+ * preview commands -
+ * formatting commands -
+ *
+ * param {vscode.ExtensionContext} the context the extension runs in, provided by vscode on activation of the extension.
+ */
+export function activate(context: ExtensionContext) {
+    extensionPath = context.extensionPath;
+    const { msTimeValue } = generateTimestamp();
+    output.appendLine(`[${msTimeValue}] - Activating docs linting extension.`);
+
+    // Markdownlint custom rule check
+    checkMarkdownlintCustomProperty();
+
+    // Update markdownlint.config to fix MD025 issue
+    addFrontMatterTitle();
+}
+
+/**
+ * Method to check for the docs custom markdownlint value.
+ * Checks for markdownlint.customRules property.  If markdownlint isn't installed, do nothing.  If markdownlint is installed, check for custom property values.
+ */
+export function checkMarkdownlintCustomProperty() {
+    const { msTimeValue } = generateTimestamp();
+    const customProperty = "markdownlint.customRules";
+    const customRuleset = "{docsmsft.docs-linting}/markdownlint-custom-rules/rules.js";
```

Comment:
Hi @adunndevster , do you think we should add logic to delete the custom ruleset entry for docs-markdown as well?  I don't think it's got any negative impact aside from noise in the output window.  Thanks

Reply:
Ah! That's what was causing that. @jamarw - you are a wizard.

Reply:
@jamarw - Okay I've updated this. The first time you run docs-linting you will see a little mess... it tries to load the old rules, is unable to, then refreshes the cache, and loads fine. After that the messages are clean up. Is this okay?

Reply:
@adunndevster Looks great!